test/browser-pool/anonymize-proxy-sugar.test.ts

```diff
@@ -0,0 +1,33 @@
+import { vi } from 'vitest';
+import { anonymizeProxy } from 'proxy-chain';
+
+import { anonymizeProxySugar } from '../../packages/browser-pool/src/anonymize-proxy';
+
+describe('anonymizeProxySugar', () => {
+    // Mock the anonymizeProxy function from proxy-chain
+    beforeEach(() => {
+        vi.mock('proxy-chain', () => ({
+            anonymizeProxy: vi.fn((url) => Promise.resolve(`anonymized-${url}`)),
+        }));
+    });
+
+    afterEach(() => {
+        vi.clearAllMocks();
+    });
+
+    test('should handle proxy url without a trailing slash correctly', async () => {
+        const proxyUrl = 'http://username:password@proxy:1000';
```

Comment:
I checked this on my phone, so take it with a grain of salt, but I think the issue only showed with `socks` proxies because of how the `new URL().href` parses different parts of URLs based on the protocol part.

![image](https://github.com/user-attachments/assets/8cbb504c-c99b-4f3e-9470-20fb5e99ca70)

Since we call `new URL()` in `anonymizeProxySugar`, this test would have passed even before this PR. We should imo test with a non-http URL here.